dwl.c

```diff
@@ -64,11 +66,15 @@ typedef struct {
 	struct wl_list link;
 	struct wl_list flink;
 	struct wl_list slink;
-	struct wlr_xdg_surface *xdg_surface;
+	union {
+		struct wlr_xdg_surface *xdg_surface;
+		struct wlr_xwayland_surface *xwayland_surface;
+	};
```

Comment:
The anonymous-union-and-a-boolean seemed the simplest approach, and sway is using it.

Happy to change if anything better is suggested.